core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ComponentVersionComparator.java

```diff
@@ -0,0 +1,24 @@
+package com.sequenceiq.cloudbreak.service.upgrade;
+
+import static com.sequenceiq.cloudbreak.cloud.PrefixMatchLength.MAJOR;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.VersionComparator;
+import com.sequenceiq.cloudbreak.cloud.VersionPrefix;
+
+@Component
+class ComponentVersionComparator {
+
+    boolean permitCmAndStackUpgradeByComponentVersion(String currentVersion, String newVersion) {
```

Comment:
Can we rename this to show when does this permit the newVersion? It's hard to figure out what actually this method does.